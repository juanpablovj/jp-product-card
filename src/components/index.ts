import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductImage } from './ProductImage';
import { ProductTitle } from './ProductTitle';
import { ProductButtons } from './ProductButtons';
import { ProductCardHOCProps } from '../interfaces/interfaces';

const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
	Title: ProductTitle,
	Image: ProductImage,
	Buttons: ProductButtons,
});

export { ProductCard, ProductImage, ProductTitle, ProductButtons };
