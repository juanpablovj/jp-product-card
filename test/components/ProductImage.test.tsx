import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('should de mostrar el componente correctamente con el img personalizado', () => {
    const wrapper = renderer.create(
        <ProductImage img={'./cat.jpg'} className='custom-class'/>
    )
    expect(wrapper).toMatchSnapshot();
  });

  test('should de mostrar el componente con el img del producto', () => { 
    const wrapper = renderer.create(
        <ProductCard product={product2} className='custom-class'>
            {() => <ProductImage />}
        </ProductCard>
    )
    expect(wrapper).toMatchSnapshot();
   })
});
