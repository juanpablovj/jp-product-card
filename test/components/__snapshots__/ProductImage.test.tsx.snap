// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should de mostrar el componente con el img del producto 1`] = `
<div
  className="productCard custom-class"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage should de mostrar el componente correctamente con el img personalizado 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src="./cat.jpg"
/>
`;
